src/main/java/use_case/search/SearchInteractor.java

```diff
@@ -0,0 +1,75 @@
+package use_case.search;
+
+import data_access.movies.MovieDBDataAccessException;
+import entity.Movie;
+import use_case.generate_recommendations.MovieDBDataAccessInterface;
+
+/**
+ * Interactor for the search by criteria use case.
+ */
+public class SearchInteractor implements SearchInputBoundary {
+    private final MovieDBDataAccessInterface movieDBDataAccess;
+    private final SearchOutputBoundary outputBoundary;
+
+    public SearchInteractor(MovieDBDataAccessInterface movieDBDataAccess, SearchOutputBoundary outputBoundary) {
+        this.movieDBDataAccess = movieDBDataAccess;
+        this.outputBoundary = outputBoundary;
```

Comment:
Consider renaming this variable to "searchPresenter" for clarity; if I didn't know clean Architecture I wouldn't really know what a class named outputBoundary does. Instead of accepting this comment (if you want to make this change) I suggest using the IDE to do a refactor (so you change all the usages at once).

```suggestion
        this.searchPresenter = searchPresenter;
```